### PR TITLE
[SDTEST-145] Support unskippable tests

### PR DESCRIPTION
**Pull request by ypopovych**
### What and why?

Added support for unskippable tests

### How?

* Added type tagging system
* Added type and instance method tags
* Changed skippable test filtering based on added tags

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

**Review by juan-fernandez — APPROVED** [review]

File: Sources/DatadogSDKTesting/DDTestObserver.swift [hunk 1]
```diff
@@ -126,9 +126,19 @@ class DDTestObserver: NSObject, XCTestObservation {
...
         if let itr = DDTestMonitor.instance?.itr {
-            let skippableTests = itr.skippableTests.filter { $0.suite == testSuite.name }.map { "-[\(testSuite.name) \($0.name)]" }
-            tests = tests.map {
-                skippableTests.contains($0.name) ? SkippedTest(for: $0) : $0
+            let skippableTests = itr.skippableTests
+            var unskippableCache: [ObjectIdentifier: UnskippableMethodChecker] = [:]
+            tests = tests.map { test in
+                guard let match = Self.testNameRegex.firstMatch(in: test.name, range: NSRange(0..<test.name.count)),
+                      let suiteRange = Range(match.range(at: 1), in: test.name),
+                      let funcRange = Range(match.range(at: 2), in: test.name) else { return test }
```
Comment (maciejburda) [inline Sources/DatadogSDKTesting/DDTestObserver.swift:134]: nit: maybe worth breaking the line after {

On the first glance I got confused with formatting

File: Sources/DatadogSDKTesting/IntelligentTestRunner/IntelligentTestRunner.swift [hunk 2]
```diff
@@ -10,18 +10,30 @@ import Foundation
...
+                    methods.get(key: test.name, or: []) { $0.append(test) }
+                }
+            }
+        }
+    }
 
     var configurations: [String: String]
     var itrFolder: Directory?
+
+    // [suite: [name: [info]]]
```
Comment (maciejburda) [inline Sources/DatadogSDKTesting/IntelligentTestRunner/IntelligentTestRunner.swift:28]: Maybe worth a dedicated model at this point?

File: Sources/DatadogSDKTesting/Utils/SwiftExtensions.swift [hunk 3]
```diff
@@ -73,26 +73,51 @@ extension Data {
         // #endif
         return result
     }
-
-    var hexString: String {
-        return self.map { String(format: "%02x", $0) }.joined()
+    
+    init?(hex: String) {
```
Comment (maciejburda) [inline Sources/DatadogSDKTesting/Utils/SwiftExtensions.swift:77]: probably worth adding dedicated tests for these two initializers

File: Sources/DatadogSDKTesting/Utils/SwiftExtensions.swift [hunk 4]
```diff
@@ -226,7 +251,39 @@ extension Dictionary where Key == String, Value == String {
 }
 
 extension Bundle {
+    @inlinable
```
Comment (maciejburda) [inline Sources/DatadogSDKTesting/Utils/SwiftExtensions.swift:254]: curious: what's the reason to use it?

File: Sources/DatadogSDKTesting/Utils/SwiftExtensions.swift [hunk 5]
```diff
@@ -226,7 +251,39 @@ extension Dictionary where Key == String, Value == String {
...
+        let value = try create()
+        self[key] = value
+        return value
+    }
+}
+
+extension Dictionary {
+    mutating func get<R>(key: Key,
+                         or create: @autoclosure () throws -> Value,
+                         _ cb: (inout Value) throws -> R) rethrows -> R
```
Comment (maciejburda) [inline Sources/DatadogSDKTesting/Utils/SwiftExtensions.swift:282]: formatting nit: not sure which formatting is more commonly followed in this repo, but I'd suggest breaking the line before the `key: Key` and after `throws -> R`

**Review by maciejburda — APPROVED** [review]
LGTM - left some minor suggestions

File: Sources/DatadogSDKTesting/Utils/SwiftExtensions.swift (hunk 3 above)
Comment (ypopovych) [reply Sources/DatadogSDKTesting/Utils/SwiftExtensions.swift:67]: yeah. good idea. added tests and refactored zlib method

File: Sources/DatadogSDKTesting/IntelligentTestRunner/IntelligentTestRunner.swift (hunk 2 above)
Comment (ypopovych) [reply Sources/DatadogSDKTesting/IntelligentTestRunner/IntelligentTestRunner.swift]: yep. Changed to the model

**Pull request merged**